File: app/src/main/java/com/vn/ezlearn/viewmodel/CustomSlideViewModel.java
package com.vn.ezlearn.viewmodel;

import android.content.Context;
import android.databinding.BaseObservable;
import android.databinding.Bindable;
import android.databinding.BindingAdapter;
import android.widget.ImageView;

/**
 * Created by FRAMGIA\nguyen.duc.manh on 13/09/2017.
 */

public class CustomSlideViewModel extends BaseObservable {
    private Context context;
    @Bindable
    public String title;
    @Bindable
    public String description;
    @Bindable
    public int image;
    @Bindable
    public int colorBg;

    public CustomSlideViewModel(Context context, String title, String description, int image,
                                int colorBg) {
        this.context = context;
        this.title = title;
        this.description = description;
        this.image = image;
        this.colorBg = colorBg;
    }

    @BindingAdapter("imageResource")
    public static void setImageResource(ImageView imageView, int resource) {
        imageView.setImageResource(resource);
    }

    @BindingAdapter("backgroundResource")
    public static void setBackgroudResource(ImageView imageView, int resource) {
        imageView.setBackgroundResource(resource);
    }
}

File: app/src/main/java/com/vn/ezlearn/viewmodel/CustomSlideViewModel.kt
package com.vn.ezlearn.viewmodel

import android.databinding.BaseObservable
import android.databinding.BindingAdapter
import android.databinding.ObservableField
import android.databinding.ObservableInt
import android.widget.ImageView

/**
 * Created by FRAMGIA\nguyen.duc.manh on 13/09/2017.
 */

class CustomSlideViewModel(var title: String, var description: String, var image: Int,
                           var colorBg: Int) : BaseObservable() {
    var mTitle: ObservableField<String> = ObservableField(title)
    var mDescription: ObservableField<String> = ObservableField(description)
    var mImage: ObservableInt = ObservableInt(image)
    var mColorBg: ObservableInt = ObservableInt(colorBg)

    companion object {
        @JvmStatic
        @BindingAdapter("imageResource")
        fun setImageResource(imageView: ImageView, resource: Int) {
            imageView.setImageResource(resource)
        }

        @JvmStatic
        @BindingAdapter("backgroundResource")
        fun setBackgroudResource(imageView: ImageView, resource: Int) {
            imageView.setBackgroundResource(resource)
        }
    }
}
